Upstream-Status: Pending [Not submitted to upstream yet]
Signed-off-by: Abdellatif El Khlifi <abdellatif.elkhlifi@arm.com>

From 28f3e8d68996ad2e3ccca45d2435b3b524daef48 Mon Sep 17 00:00:00 2001
From: Vishnu Banavath <vishnu.banavath@arm.com>
Date: Sat, 13 Nov 2021 07:51:53 +0000
Subject: [PATCH] fix EARLY_TA_PATHS env variable

Yocto cleans up environment varaibles at build time.
EARLY_TA_PATHS should be set a separate rule for securepartitions
to be included into optee-os image

Signed-off-by: Vishnu Banavath <vishnu.banavath@arm.com>

diff --git a/environments/opteesp/sp.mk.in b/environments/opteesp/sp.mk.in
index c44ad59..d67e2dc 100644
--- a/environments/opteesp/sp.mk.in
+++ b/environments/opteesp/sp.mk.in
@@ -14,7 +14,8 @@ ifeq (,${@EXPORT_SP_UUID@-included})
 endif
 
 ifeq (embedded,${SP_PACKAGING_METHOD})
-OPTEE_OS_COMMON_EXTRA_FLAGS+=EARLY_TA_PATHS+=${TS_INSTALL_PREFIX}/opteesp/bin/@EXPORT_SP_UUID@.stripped.elf
+EARLY_TA_PATHS+=${TS_INSTALL_PREFIX}/bin/@EXPORT_SP_UUID@.stripped.elf
+OPTEE_OS_COMMON_EXTRA_FLAGS+=${EARLY_TA_PATHS}
 TS_SP_DTSI_LIST+="\\n\#include \"${TS_INSTALL_PREFIX}/opteesp/manifest/@EXPORT_SP_UUID@.dtsi\""
 else ifeq (fip,${SP_PACKAGING_METHOD})
 TS_SP_JSON_LIST+=${TS_INSTALL_PREFIX}/opteesp/json/@EXPORT_SP_NAME@.json
-- 
2.17.1

